src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -248,6 +246,65 @@ final case class EServicesApiServiceImpl(
     draftDescriptor = getDraftDescriptor(eService).map(_.toCompactDescriptor)
   )
 
+  override def getProducerEServiceDescriptor(eserviceId: String, descriptorId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerProducerDescriptor: ToEntityMarshaller[ProducerEServiceDescriptor]
+  ): Route = {
+    val result: Future[ProducerEServiceDescriptor] = for {
+      requesterId                    <- getOrganizationIdFutureUUID(contexts)
+      (eserviceUUID, descriptorUUID) <- eserviceId.toFutureUUID.zip(descriptorId.toFutureUUID)
+      eService                       <- catalogProcessService.getEServiceById(eserviceUUID)
+      _                              <- isTheProducer(eService, requesterId)
```

Comment:
Captain Obvious 🎖️  here: this is an antipattern, even though I do not currently see any alternative

Reply:
let's say we bend this authorization not to the data returned but the access to the endpoint, and leave it as is